remove some uneeded sleeps in tests

cpuguy83 opened the pull request.

calavera committed:
Shallow clone using git to build images.

Signed-off-by: David Calavera <david.calavera@gmail.com>

calavera committed:
Do not try to shallow git history when the protocol doesn't allow it.

This only happens with the old git http dumb protocol, but that's what we use in our integration tests.
We check the Content-Type header advertised in http requests to make sure the http transport is the git smart transport:

See this commit as a reference:
https://github.com/git/git/commit/4656bf47fca857df51b5d6f4b7b052192b3b2317

Signed-off-by: David Calavera <david.calavera@gmail.com>

calavera committed:
Test that we set the right arguments for git cloning depending on the transport.

Signed-off-by: David Calavera <david.calavera@gmail.com>

LK4D4 commented on integration-cli/docker_api_logs_test.go:
```diff
@@ -5,26 +5,46 @@ import (
 func (s *DockerSuite) TestLogsApiWithStdout(c *check.C) {
-	name := "logs_test"
+	name := "log_test"
+	out, _ := dockerCmd(c, "run", "-d", "-t", "busybox", "/bin/sh", "-c", fmt.Sprintf("trap 'echo %s' SIGTERM; while true; do sleep 0.01; done", name))
```
Why not just `top` instead of loop?

cpuguy83 replied:
Because top was exiting with "143" when I sigterm'd

vbatts committed:
contrib/init: unshare mount namespace for inits

* openrc
* sysvinit-debian
* upstart

Signed-off-by: Vincent Batts <vbatts@redhat.com>

calavera committed:
Document the extra `depth` argument in git contexts.

Signed-off-by: David Calavera <david.calavera@gmail.com>

calavera committed:
Remove duplicated git clone logic.

Signed-off-by: David Calavera <david.calavera@gmail.com>

DarianAnjuhal committed:
fix runtime issue for TestEventsFilterContainer

Signed-off-by: Daniel Antlinger <d.antlinger@gmx.at>
Signed-off-by: Alexander Morozov <lk4d4@docker.com>

LK4D4 committed:
Implement teardown removeAllImages

Signed-off-by: Alexander Morozov <lk4d4@docker.com>

jfrazelle commented:
ugh rebase sorry :(

cpuguy83 commented:
rebased

HuKeping committed:
Add test for REST API container rename

Signed-off-by: Hu Keping <hukeping@huawei.com>

vdemeester committed:
Add coverage on pkg/archive

Add tests on:
- changes.go
- archive.go
- wrap.go

Should fix #11603 as the coverage is now 81.2% on the ``pkg/archive``
package. There is still room for improvement though :).

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

LK4D4 commented on integration-cli/docker_api_logs_test.go:
```diff
@@ -5,24 +5,44 @@ import (
 	"github.com/go-check/check"
 )
 
 func (s *DockerSuite) TestLogsApiWithStdout(c *check.C) {
```
It is still took 11 seconds on jenkins. Do you have any idea why?

cpuguy83 replied:
Sure do... loop isn't exiting on SIGTERM :) Fix incoming.

cpuguy83 replied:
Updated, tests good. 0.61s now.

LK4D4 committed:
Merge pull request #12502 from calavera/shallow_git_history

Shallow clone using git to build images.

LK4D4 commented:
LGTM

LK4D4 committed:
Use -f for rm instead of kill before

Signed-off-by: Alexander Morozov <lk4d4@docker.com>

duglin committed:
Merge pull request #12753 from LK4D4/remove_all_images

Implement teardown removeAllImages

cpuguy83 committed:
Merge pull request #12754 from LK4D4/carry_12707

fix runtime issue for TestEventsFilterContainer

ibuildthecloud committed:
Expose ParseRestartPolicy

ParseRestartPolicy is useful function for third party go programs to use
so that they can parse the restart policy in the same way that Docker
does

Signed-off-by: Darren Shepherd <darren@rancher.com>

cpuguy83 committed:
Merge pull request #12755 from LK4D4/use_f_for_remove

Use -f for rm instead of kill before

LK4D4 committed:
Not protect dangling images for integration-cli

Signed-off-by: Alexander Morozov <lk4d4@docker.com>

cpuguy83 committed:
Merge pull request #12762 from ibuildthecloud/export-restart-parse

Expose ParseRestartPolicy

LK4D4 committed:
Implement DockerRegistrySuite in integration-cli

To avoid manually creating and destroying registrys in tests.

Signed-off-by: Alexander Morozov <lk4d4@docker.com>

LK4D4 committed:
Merge pull request #12764 from LK4D4/not_protect_dangling

Not protect dangling images for integration-cli

Commit:
Fixed typo 'configuring'

Signed-off-by: John Howard <jhoward@microsoft.com>

Commit:
Merge pull request #12775 from Microsoft/12774-fixtypo

Fixed typo 'configuring'

Commit:
Merge pull request #12699 from vbatts/vbatts-init_mnt_ns

contrib/init: unshare mount namespace for inits

estesp committed:
Merge pull request #12765 from LK4D4/registry_suite

Implement DockerRegistrySuite in integration-cli

LK4D4 committed:
Merge pull request #12744 from vdemeester/11603-pkg-archive-test-coverage-2

Add coverage on pkg/archive (related to #11603)

sergey-alekseev committed:
Fix documentation typo in articles/basics.md

Signed-off-by: Sergey Alekseev <sergey.alekseev.minsk@gmail.com>

Commit:
A fix for = in env values in linked containers

Closes: #12763

Signed-off-by: Doug Davis <dug@us.ibm.com>

Commit:
Fix race condition in API commit test

Signed-off-by: Doug Davis <dug@us.ibm.com>

LK4D4 committed:
Merge pull request #12784 from duglin/RaceCommit

Fix race condition in API commit test

epc committed:
Copy edits for typos

Signed-off-by: Ed Costello <epc@epcostello.com>

LK4D4 committed:
Merge pull request #12728 from HuKeping/addtest

Add test for REST API container rename

calavera committed:
Merge pull request #12734 from sergey-alekseev/sergey-alekseev-patch-1

Fix documentation typo in articles/basics.md

Commit:
Merge pull request #12787 from epc/copyedits

Copy edits for typos

duglin commented on integration-cli/docker_api_logs_test.go:
```diff
@@ -5,24 +5,44 @@ import (
+		chLog <- logOut{body, statusCode, err}
+	}()
+
+	time.Sleep(100 * time.Millisecond)
```
You sure this isn't racey? What if it takes more than 100 milliseconds for the GET to get to the deamon?

cpuguy83 replied:
Technically yes it is inherently racey... do you see a good way around this?

duglin replied:
Could we have the container print "name" to the screen, over and over. And then wait for it to show up in the logs?

cpuguy83 replied:
That's a pretty big change to this test... I'm not sure why it's doing what it's doing... let me do a git blame and see where it came in.

cpuguy83 replied:
Actually, this was moved from server_unit tests... I think we can make this change.

duglin replied:
I _think_ its just checking to see that we see "name" in the logs and the sleep 10 was there to give the "docker logs" time to get started because if it started to see then we'd miss the one and only println. So having lots of println should be ok - I think.

cpuguy83 replied:
Updated this one now that I saw the original intention of the test.  No more racey's here.

duglin commented on integration-cli/docker_cli_wait_test.go:
```diff
@@ -44,19 +44,29 @@ func (s *DockerSuite) TestWaitNonBlockedExitZero(c *check.C) {
 
-	if err != nil || strings.TrimSpace(out) != "0" {
-		c.Fatal("failed to set up container", out, err)
+	time.Sleep(100 * time.Millisecond)
```
This one is racey too much I can't think of a good way to know when the 'wait' is there and blocking, so until someone else can think of a better way, this is ok.

cpuguy83 replied:
Yeah, and issue with it before was that if there was an issue with wait, it'd just hang and never timeout... which we could still do.

duglin commented:
LGTM

duglin committed:
Merge pull request #12616 from cpuguy83/remove_uneeded_sleeps

remove some uneeded sleeps in tests

duglin commented:
so close - guess a rebase is needed

cpuguy83 committed:
Merge pull request #12783 from duglin/Issue12763-LinkedEnvs

A fix for = in env values in linked containers

GordonTheTurtle commented:
Can you please sign your commits following these rules:
https://github.com/docker/docker/blob/master/CONTRIBUTING.md#sign-your-work
The easiest way to do this is to amend the last commit:
~~~console
$ git clone -b "remove_uneeded_sleeps" git@github.com:cpuguy83/docker.git somewhere
$ cd somewhere
$ git rebase -i HEAD~35
editor opens
change each 'pick' to 'edit'
save the file and quit
$ git commit --amend -s --no-edit
$ git rebase --continue # and repeat the amend for each commit
$ git push -f
~~~

This will update the existing PR, so **DO NOT** open a new one.

cpuguy83 commented:
FYI, this was merged but due to some weird UI sync bug with GH the PR itself got closed...and commits shown all jacked up after force-pushing to this brach (b/c the UI looked like there was a merge conflict, but there wasn't) after GH had had already merged the actual commit... and when this branch was force pushed GH automatically closed this PR.... weird stuff...

TLDR; this is merged.
